Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1683.cs

```diff
@@ -0,0 +1,202 @@
+				layout.Children.Add(new Label() { Text = "No Keyboard will use Sentence" });
+			}
+			else if (Device.RuntimePlatform == Device.Android)
+			{
+				layout.Children.Add(new Label() { Text = "All will use Sentence" });
+				layout.Children.Add(new Label() { Text = "No Keyboard will use None" });
+			}
+
+			foreach (InputView child in inputViews)
+			{
+				var inputs = new StackLayout()
+				{
+					Orientation = StackOrientation.Horizontal
+				};
+
+				if (child is Entry)
+					(child as Entry).Text = "All the Same.";
+
+				if (child is Editor)
+					(child as Editor).Text = "All the Same.";
+
+
+				child.HorizontalOptions = LayoutOptions.FillAndExpand;
+				var theLabel = new Label();
+
+				theLabel.SetBinding(Label.TextProperty, new Binding("ClassId", source: child));
+				inputs.Children.Add(theLabel);
+				inputs.Children.Add(child);
+
+				layout.Children.Add(inputs);
+			}
+
+
+
+			Button rotate = new Button()
+			{
+				Text = "Change Capitalization Settings. Ensure they update correctly",
+				AutomationId = "Rotation"
+			};
+
+
+			// This shifts everyones capitalization by one in order
+			// to test that updating the field works as expected
+			rotate.Clicked += (_, __) =>
+			{
+				var item1 = entryViews[0];
+				entryViews.Remove(item1);
+				entryViews.Add(item1);
+
+				var item2 = editorViews[0];
+				editorViews.Remove(item2);
+				editorViews.Add(item2);
+
+				for (int i = 0; i <= flags.Length; i++)
+				{
+					var editorView = editorViews[i] as EditorKeyboardFlags;
+					var entryView = entryViews[i] as EntryKeyboardFlags;
+
+					if (i == flags.Length)
+					{
+						entryView.FlagsToTestFor = null;
+						entryView.Keyboard = null;
+						entryView.ClassId = "EntryNoKeyboard";
+
+
+						editorView.FlagsToTestFor = null;
+						editorView.ClassId = "EntryNoKeyboard";
+						editorView.Keyboard = null;
+					}
+					else
+					{
+						entryView.FlagsToTestFor = flags[i];
+						entryView.Keyboard = Keyboard.Create(flags[i] | spellCheckForUwp);
+						entryView.ClassId = $"Entry{flags[i]}";
+
+						editorView.FlagsToTestFor = flags[i];
+						editorView.Keyboard = Keyboard.Create(flags[i] | spellCheckForUwp);
+						editorView.ClassId = $"Editor{flags[i]}";
+					}
+				}
+			};
+
+			StackLayout content = new StackLayout();
+			content.Children.Add(new ScrollView()
+			{
+				Content = layout
+			});
+
+			content.Children.Add(rotate);
+
+			Content = content;
+		}
+
+
+
+		public class EditorKeyboardFlags : Editor
+		{
+			public KeyboardFlags? FlagsToSet { get; set; }
+
+
+			public KeyboardFlags? FlagsToTestFor { get; set; }
+		}
+
+
+		public class EntryKeyboardFlags : Entry
+		{
+			public KeyboardFlags? FlagsToSet { get; set; }
+
+
+			public KeyboardFlags? FlagsToTestFor { get; set; }
+		}
+
+
+#if UITEST
+		[Test]
+		public void Issue1683Test()
+		{
+			for (int i = 0; i < 6; i++)
+			{
+				RunningApp.WaitForElement("Rotation");
```

Comment:
`RunningApp.WaitForElement(q => q.Marked("Rotation")); ` should fix your UI test errors